daal4py/sklearn/model_selection/split.py

```diff
@@ -0,0 +1,149 @@
+    shuffle = options.pop('shuffle', True)
+    rng = options.pop('rng', 'MT19937')
+
+    available_rngs = ['MT19937', 'SFMT19937', 'MT2203', 'R250', 'WH', 'MCG31',
+                      'MCG59', 'MRG32K3A', 'PHILOX4X32X10', 'NONDETERM']
+    if rng not in available_rngs:
+        raise ValueError(
+            "Wrong random numbers generator is chosen. "
+            "Available generators: %s" % str(available_rngs)[1:-1])
+
+    if options:
+        raise TypeError("Invalid parameters passed: %s" % str(options))
+
+    arrays = indexable(*arrays)
+
+    n_samples = _num_samples(arrays[0])
+    n_train, n_test = _validate_shuffle_split(n_samples, test_size, train_size,
+                                              default_test_size=0.25)
+    if shuffle is False:
+        if stratify is not None:
+            raise ValueError(
+                "Stratified train/test split is not implemented for "
+                "shuffle=False")
+
+        train = np.arange(n_train)
+        test = np.arange(n_train, n_train + n_test)
+    else:
+        if stratify is not None:
+            cv = StratifiedShuffleSplit(test_size=n_test,
+                                        train_size=n_train,
+                                        random_state=random_state)
+            train, test = next(cv.split(X=arrays[0], y=stratify))
+        else:
+            if mkl_random_is_imported and (isinstance(random_state, int) or random_state is None):
+                random_state = mkl_random.RandomState(random_state, rng)
+                indexes = random_state.permutation(n_train + n_test)
+                train, test = indexes[:n_train], indexes[n_train:]
+            else:
+                cv = ShuffleSplit(test_size=n_test,
+                                  train_size=n_train,
+                                  random_state=random_state)
+                train, test = next(cv.split(X=arrays[0], y=stratify))
+
+    res = []
+    for arr in arrays:
+        fallback = False
+        if not isinstance(arr, np.ndarray):
+            if pandas_is_imported:
+                if not isinstance(arr, pd.core.frame.DataFrame) and not isinstance(arr, pd.core.series.Series):
+                    fallback = True
+            else:
+                fallback = True
+
+        dtypes = get_dtypes(arr)
+        if dtypes is None:
+            fallback = True
+        else:
+            for i, dtype in enumerate(dtypes):
+                if dtype not in [np.int32, np.float32, np.float64, 'int32', 'float32', 'float64']:
+                    fallback = True
+                    break
+        if fallback:
+            train = train.reshape((n_train,))
+            test = test.reshape((n_test,))
+            res.append(_safe_indexing(arr, train))
+            res.append(_safe_indexing(arr, test))
+        else:
+            train = train.reshape(n_train, 1)
+            test = test.reshape(n_test, 1)
+
+            if len(arr.shape) == 2:
+                one_dim = False

```

Comment:
```one_dim``` is never used. So, I think it should be removed

Reply:
deleted, thanks for reminder